lib/nsq/connection.rb

```diff
@@ -32,6 +32,7 @@ def initialize(opts = {})
       @msg_timeout = opts[:msg_timeout] || 60_000 # 60s
       @max_in_flight = opts[:max_in_flight] || 1
       @ssl_context = opts[:ssl_context]
+      @tls_v1 = !!opts[:tls_v1]
       validate_ssl_context! if @ssl_context
```

Comment:
Warning sounds like a nice thing to do in that somewhat odd case.
Maybe the `@tls_v1` check and `warn` could go in the `validate_ssl_context!` method like so:

```
def validate_ssl_context!
  unless @tls_v1
    warn 'tls_v1 is false but a context is set: validation skipped'
    return
  end
  ...
end
```

Not 100% sure that is the correct place for it but thinking out loud.